src/github.com/getlantern/flashlight/geolookup/geolookup.go

```diff
@@ -88,6 +91,45 @@ func registerService() error {
 	return err
 }
 
+func lookupIp(httpClient *http.Client) (string, string, error) {
+	httpClient.Timeout = 60 * time.Second
+
+	var err error
+	var req *http.Request
+	var resp *http.Response
+
+	if req, err = http.NewRequest("HEAD", "http://nl.fallbacks.getiantem.org", nil); err != nil {
+		return "", "", fmt.Errorf("Could not create request: %q", err)
+	}
+
+	// Enproxy returns an error if this isn't there.
+	req.Header.Set(enproxy.X_ENPROXY_ID, "1")
+
+	if resp, err = httpClient.Do(req); err != nil {
+		return "", "", fmt.Errorf("Could not get response from server: %q", err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			log.Debugf("Unable to close reponse body: %v", err)
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		body := "body unreadable"
+		b, err := ioutil.ReadAll(resp.Body)
+		if err == nil {
+			body = string(b)
+		}
+		return "", "", fmt.Errorf("Unexpected response status %d: %v", resp.StatusCode, body)
+	}
+
+	ip := resp.Header.Get("Lantern-Ip")
```

Comment:
We are 100% positive that Enproxy will respond with these headers right? Even though a bit pedantic, wouldn't a test to make sure be a good idea? One rename or refactor could sneak empty strings here.